### ISSUE TITLE
Normalize flag in WindowSamplingConfig

### ISSUE THREAD

**Issue by bdevoghel** [azavea/raster-vision#2245]
## 🚀 Feature
When creating a `SemanticSegmentation**GeoDataConfig**` (maybe other use-cases than segmentaion too ? my experience is limited to this one), the `scene_to_dataset` creates a `GeoDataset` based on options from the sampling config. As `WindowSamplingConfig` has no `normalize` flag, the created dataset can only be created with `normalize=True`; which is undesired.

The requested feature is to create the `normalize` flag in the `WindowSamplingConfig` class and passing it down where required.

## Motivation

When creating a `SemanticSegmentation**GeoDataConfig**`, I want my dataset to be sampled without built-in normalization.

## Pitch

The class `WindowSamplingConfig` should have the additional `normalize` pydantic field.

## Alternatives

Creating my own custom class similar to `SemanticSegmentationGeoDataConfig` but with the enforces `normalize` flag set to false.

**Comment by AdeelH** [comment]
Thanks for the issue and the PR! Can you say a bit more about your use case? Do you want to, say, feed in `uint16` values type-casted directly to floats to the model? One existing way to do that is to attach a `CastTransformer` to your `RasterSource`. E.g.:

```py
from rastervision.core.data import CastTransformerConfig

raster_source = RasterioSourceConfig(..., transformers=[CastTransformerConfig(to_dtype='float32')])
```

**Comment by AdeelH** [comment]
Right, I understand. What I meant was that that normalization only happens if the data is an unsigned integer type, so if the data is already type-casted to a float type (via the `CastTransformer`), it would not be affected even if `normalize=True`.

That said, I think exposing `normalize` via a `Config` makes sense. But I don't think `WindowSamplingConfig` is the best place for it, since that is supposed to be concerned only with the sampling of windows and not the transformation of data. What I would suggest is to:
- Add it [`DataConfig`](https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py#L614). That way it will be available in both `GeoDataConfig`s and `ImageDataConfig`, both of which are based on `AlbumentationsDataset`. This will also require modifying `dir_to_dataset`s in `*ImageDataConfig`s, but feel free to skip that in this PR.
- Rename the field to `normalize_uint` (only in the `Config`s and not the `Dataset`s) to better express what it is for.

Please also remember to [sign the CLA](https://docs.rastervision.io/en/stable/CONTRIBUTING.html).
